numpy/lib/function_base.py

```diff
@@ -1557,28 +1557,59 @@ def sort_complex(a):
         return b
 
 
-def _trim_zeros(filt, trim=None):
+def _trim_zeros(
+    filt,
+    trim=None,
+    axis=None,
+    *,
+    atol=None,
+    rtol=None,
+    return_lengths=None
+):
     return (filt,)
 
 
 @array_function_dispatch(_trim_zeros)
-def trim_zeros(filt, trim='fb'):
-    """
-    Trim the leading and/or trailing zeros from a 1-D array or sequence.
+def trim_zeros(
+    filt,
+    trim='fb',
+    axis=-1,
+    *,
+    atol=0,
+    rtol=0,
+    return_lengths=False
+):
+    """Remove values along a dimension which are zero along all other.
 
     Parameters
     ----------
-    filt : 1-D array or sequence
+    filt : array_like
         Input array.
     trim : str, optional
         A string with 'f' representing trim from front and 'b' to trim from
-        back. Default is 'fb', trim zeros from both front and back of the
-        array.
+        back. By default, zeros are trimmed from the front and back.
+    axis : int or sequence, optional
+        The axis or a sequence of axes to trim. If None all axes are trimmed.
+    atol : float, optional
+        Absolute tolerance with which a value is considered for trimming.
+    rtol : float, optional
+        Relative tolerance with which a value is considered for trimming.
```

Comment:
I don't really like these arguments here - it seems like you're after a function for replacing almost-zeros with zeros, which can then be composed, as `trim_zeros(zero_small_values(x, rtol=...), axis=1)` or similar.

Reply:
I don't mind removing them. I added them for convenience but it's easy to assign zeros appropriately before calling this function. 

Seeing that there is already [numpy.isclose](https://docs.scipy.org/doc/numpy/reference/generated/numpy.isclose.html) I'm not convinced that a another function is needed.